pkg/crc/machine/start.go

```diff
@@ -793,7 +793,7 @@ func updateKubeconfig(ctx context.Context, ocConfig oc.Config, sshRunner *crcssh
 }
 
 func bundleMismatchWithPreset(preset crcPreset.Preset, bundleMetadata *bundle.CrcBundleInfo) error {
-	if preset != crcPreset.OpenShift && bundleMetadata.IsOpenShift() {
+	if preset == crcPreset.Podman && bundleMetadata.IsOpenShift() {
```

Comment:
Arguably that's `bundleMetadata.IsOpenShift()` which is buggy, it should return false for an OKD bundle. 
I think we have no way of detecting this mistake:
```
crc config set preset okd
crc start --bundle crc_libvirt_4.10.14.crcbundle
```
with `crc_libvirt_4.10.14.crcbundle` being an OpenShift (not OKD) bundle. This will skip the pull secret and will fail to start.
I'd add a few more details to the commit log as this is not fully obvious what's broken/why we don't add more mistmatch detection now that we have more presets.

Reply:
Right, as of now bundle created from snc (ocp/okd) have type as `snc` which is correct for (single node cluster) both the cases. We do need to figure out best way to detect if bundle is OCP/OKD and print out the mismatch here. May be add another field in bundle-info data (snc side) as `preset` and put `ocp/okd/podman` according to bundle generation. As of now it is half backed solution (buggy one) but good as long as user manually download bundle and use it. I will put this details in the commit log and then start putting stuff on snc side.